src/misc/guillotine.ts

```diff
@@ -0,0 +1,197 @@
+  BlockPistonRetractEvent,
+} from 'org.bukkit.event.block';
+import { PlayerInteractEvent } from 'org.bukkit.event.player';
+import { EquipmentSlot, ItemStack } from 'org.bukkit.inventory';
+import { SkullMeta } from 'org.bukkit.inventory.meta';
+import { spawnHolderArmorStand } from '../common/entities/armor-stand';
+import { isRightClick } from '../common/helpers/click';
+import { CustomItem } from '../common/items/CustomItem';
+import { VkItem } from '../common/items/VkItem';
+
+const DEADLY_CHOP_VELOCITY = 0.6; // Approx. > 3 block fall kills always.
+const VEL_DAMAGE_MODIFIER = 12; // This is multiplied with y_velocity to get smaller drop damage.
+
+const PLACEMENT_SOUND = 'minecraft:block.anvil.place';
+const LAND_SOUND = 'minecraft:block.anvil.land';
+
+const cooldowns = new Set<Entity>();
+
+const Blade = new CustomItem({
+  id: 5,
+  name: translate('vk.guillotine_blade'),
+  type: VkItem.UNSTACKABLE,
+});
+
+// prettier-ignore
+const HEADS = new Map<EntityType, ItemStack>([
+  [EntityType.ZOMBIE,           new ItemStack(Material.ZOMBIE_HEAD, 1)],
+  [EntityType.CREEPER,          new ItemStack(Material.CREEPER_HEAD, 1)],
+  [EntityType.CREEPER,          new ItemStack(Material.CREEPER_HEAD, 1)],
+  [EntityType.SKELETON,         new ItemStack(Material.SKELETON_SKULL, 1)],
+  [EntityType.WITHER_SKELETON,  new ItemStack(Material.WITHER_SKELETON_SKULL, 1)],
+]);
+
+async function dropBlade(armorStand: ArmorStand) {
+  if (armorStand.hasGravity()) return; // Has gravity -> Blade already falling.
+  // Begin drop
+  armorStand.setGravity(true);
+  armorStand.setInvulnerable(true);
+  let previousY = armorStand.velocity.y;
+
+  // Blade falling
+  for (let i = 0; i < 200; i++) {
+    await wait(2, 'ticks');
+    const currentY = armorStand.velocity.y;
+    if (!armorStand.isValid() || currentY >= previousY) break;
+    previousY = currentY;
+    chopIfHit(armorStand.location, Math.abs(currentY));
+  }
+
+  // Blade hit ground
+  armorStand.world.playSound(
+    armorStand.location,
+    LAND_SOUND,
+    SoundCategory.BLOCKS,
+    0.6,
+    0.8,
+  );
+
+  armorStand.setGravity(false);
+  armorStand.setInvulnerable(false);
+}
+
+async function chopIfHit(location: Location, y_velocity: number) {
+  const entities = location.getNearbyEntities(1, 1.2, 1);
+  for (const entity of entities) {
+    if (!(entity instanceof LivingEntity)) return;
+    if (entity instanceof ArmorStand) return;
+    // Set cooldown
+    if (cooldowns.has(entity)) return;
+    cooldowns.add(entity);
+
+    // Wait for better timing
+    await wait(4, 'ticks');
+
+    // Deal damage
+    if (y_velocity >= DEADLY_CHOP_VELOCITY) entity.health = 0;
+    else entity.damage(VEL_DAMAGE_MODIFIER * y_velocity);
+
+    // Play effects
+    const eyeLocation = entity.eyeLocation;
+    playChopEffects(eyeLocation);
+
+    // Drop head
+    if (entity.isDead()) dropHead(entity, eyeLocation);
+
+    // Delete cooldown
+    await wait(20, 'ticks');
+    cooldowns.delete(entity);
```

Comment:
This will pause the loop for 20 ticks? What if there are multiple entities under the blade? Is it going to take multiple seconds to chop all heads?